dbcon/execplan/arithmeticoperator.h

```diff
@@ -238,6 +239,51 @@ inline void ArithmeticOperator::evaluate(rowgroup::Row& row, bool& isNull, Parse
       break;
 
     case execplan::CalpontSystemCatalog::UBIGINT:
+      {
+        // XXX: this is bandaid solution for specific customer case (MCOL-5568).
+        // Despite that I tried to implement a proper solution: to have operations
+        // performed using int128_t amd then check the result.
+        int128_t x, y;
```

Comment:
We have TSUInt64 type that I suggest to improve with math result overflow detection so the code can be reused in future if needed.